### PR TITLE
Exclude @keyframes params from being parsed as a Less variable

### LINKED ISSUE #9339: @keyframes rule with global scope in CSS modules produces invalid CSS
**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABA1nAngMwE4CGAtnAM4AESA5gDYQBGBNAFKQA4CWUAlOcADpRy5fBCJ9Bw4TEJRSOCHiJJyeCDALxmABgAmcKtwDck8gF9TMCBKFSZBOQqUq1GrQGYAbHoPHTFqDMQABoQCDYYDmhSZFACPDUAdwAFeIQYlCZEgiwY0PpCMEwYAGU2AjAuKmQZAFc4ULgiejhdfV0AGQcqWoIqOAAxRSJNSKhqzNqrEJAACxgiGgB1WY54dgq4EvS1jgA3NaxkcFI8kC5SODwYZMIqEeQcJkvQgCtSAA8AIULikuI4B0uHBHs8GiB3h8SlUaHAAIq1dQgpBPGgvEDlPCXPDHMCnGZsPBcGBLDi6GCzZAADm0oUJEEuS0IbGOhLIVz2INCAEdEfBbuEMiACKQALRQOCtVozPBwXkcWW3PoPFFg0KXIgcGp4erqmHwvnI1HojT0UnkylIABMoXsHBoVQAwmIVSAyABWGa1S4AFQI9AyxvBe3qAEkoPpYCUwESIgBBCMlGBYWGgtFwMxmIA)
```sh
--parser css
```

**Input:**
```css
@keyframes :global(spin) {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
```

**Output:**
```css
@keyframes: global(spin){
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
};

```

**Expected behavior:**
```css
@keyframes :global(spin) {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

### PR DESCRIPTION
**Pull request by tmazeika**
## Description
Fixes #9339.

The at-rule parameters for `@page` and `@nest` are safe from having their leading `:` moved to `node.raws.afterName`, but not `@keyframes`, causing it to be treated like a Less variable when one uses the CSS modules `:global(...)` and `:local(...)` scope modifiers. That then causes `@keyframes` blocks to be prettified to invalid CSS.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Can you fix the spellcheck? Or you can allow "maintainers to edit"(I failed to push to your fork).

File: changelog_unreleased/css/10773.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#### Exclude @keyframes params from being parsed as a Less variable (#10773 by @tmazeika)
+
+<!-- prettier-ignore -->
+```css
+/* Input */
+@keyframes: global(spin) {
```
Comment (fisker) [inline changelog_unreleased/css/10773.md:6]:
```suggestion
@keyframes :global(spin) {
```

File: changelog_unreleased/css/10773.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#### Exclude @keyframes params from being parsed as a Less variable (#10773 by @tmazeika)
```
Comment (fisker) [inline changelog_unreleased/css/10773.md:1]:
```suggestion
#### Exclude `@keyframes` params from being parsed as a Less variable (#10773 by @tmazeika)
```

**Review by fisker — CHANGES_REQUESTED** [review]

**Comment by tmazeika** [comment]
> Can you fix the spellcheck? Or you can allow "maintainers to edit"(I failed to push to your fork).

Hmm, maintainer edits should be allowed... I've committed your changes, but I'm still getting `yarn lint:spellcheck` errors. It's fixed if I add `<!-- cspell:disable-next-line -->` at line 1 in 10773.md, but I'm not sure if that's desired. Cheers.

**Comment by fisker** [comment]
Just add `keyframes` to  https://github.com/prettier/prettier/blob/main/cspell.json

**Comment by tmazeika** [comment]
Should be good to go, thanks.

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Thank you! We'll need more approves from other maintainers.

**Review by thorn0 — APPROVED** [review]

**Pull request merged**